### ISSUE TITLE
Disable editing of Project Prefs > Project Base Dir field

### ISSUE THREAD

**Issue by cgchoffman** [Komodo/KomodoEdit#2155]
### Short Summary

Komodo will open a remote file and show the wrong protocol in the open file's path if you use the Projects base dir field to manually enter the wrong protocol.

### Steps to Reproduce
Prerequisites:
This requires that you have a server that supports one protocol but not both of either `sftp` or `ftp`
- Set up an connection trough sftp ( server )
- Create a project with a remote url to the server connection you just created 
  - Project Prefs > Project Base Dir field > Browse remote
- Refresh you're project by double clicking on it
- open a file from the server (note that it used the protocol you set in your server settings)
- Open project settings and manually change the url from `sftp` to `ftp` (or the opposite of what ever you chose in the first place) and save the settings
  - Do **not** use the Browse Remote button
- Refresh you're project by double clicking on it
- Open the same file

### Expected results

I'm not even really sure what I would expect to happen...I guess I would expect Komodo to do nothing and fail to open the file as seemingly i'm asking it to open a file on a protocol that my server doesn't support.

### Actual results

A second instance of the file is opened and the PATH to that file will show the new manually set protocol.

Long story short, that field should just not be editable.  That would be frustrating for local paths, but for remote paths it makes sense.  If we were to try and keep it editable, we could change that field to **only** show the path, NOT the protocol and server connection pref used.

**Comment by cgchoffman** [comment]
This COULD be causing issues such as #1758.  If that wrong path is being saved in global prefs or a remote files MRU then it could reek havoc in Komodo else where.

**Comment by we3geeks** [comment]
Based on the discussion at http://community.komodoide.com/t/komodo-edit-how-to-put-files-into-a-project/2682 I'm concerned about the impacts of not allowing the project base directory to be editable. 

I've used that capability to good advantage in situations where I have projects that consist of a primary folder structure, plus one or more secondary folder structures outside of that primary, and I don't want to have the project file itself (the .komodoproject file) in the primary folder for source code control (or other) reasons.

**Comment by cgchoffman** [comment]
You would still be able to edit the field through the "local" and "remote" buttons.  There is no suggestion here to take away functionality.  Your use case would still be supported.

Since you have mutliple folders need to be included in your project, it actually sounds like you should use the "add existing folder" functionality.

**Comment by Naatan** [comment]
Fwiw the exact UX is still tbd here, the main goal of this issue is to ensure the field doesnt allow you to input any invalid values.

**Comment by we3geeks** [comment]
@cgchoffman and @Naatan Thanks for the clarifications; it would seem that I read more into the "that field should not be editable" phrasing of the original description than perhaps I should have.

As long as my use-case (.komodoproject outside of the base directory, base directory is the primary folder structure, one or more additional directory structures added to the project via "Add Existing Folder..."), I'll be happy.

**Comment by Naatan** [comment]
Absolutely, that is my use case also.
